### ISSUE TITLE
RecordNotUnique for BgsPowerOfAttorney in Jobs

### ISSUE THREAD

**Issue by ferristseng** [department-of-veterans-affairs/caseflow#14509]
## Description

Error:

```
ActiveRecord::RecordNotUnique: PG::UniqueViolation: ERROR:  duplicate key value violates unique constraint "bgs_poa_pid_fn_unique_idx"
```

- https://sentry.ds.va.gov/department-of-veterans-affairs/caseflow/issues/10410/?referrer=webhooks_plugin
- https://sentry.ds.va.gov/department-of-veterans-affairs/caseflow/issues/10307/?referrer=webhooks_plugin

This is happening in: `UpdateCachedAppealsAttributesJob` and `WarmBgsCachesJob`

## Acceptance criteria
- [ ] Fix bug

## Background/context/resources

Sentry alert

**Comment by pkarman** [comment]
Seems to be happening because there are 2 rows: one with a `nil` file number and one with file number not nil.

The reason we did not treat PID as unique by itself at the db layer was because we were pretty sure there might be multiple file numbers per-PID (like there are for Veterans). But maybe we need to add some logic to only save a new POA record is one does not already exist for *either* pid or filenumber.

**Comment by pkarman** [comment]
Correction: in at least some cases there are multiple rows with same PID and different file number (as we expected) but bug seems to be like we are trying to update a row with filenumber A and it conflicts with existing record for filenumber B. Perhaps we should delete all and insert rather than update.

Looks like there are 22 of these situations.

```
BgsPowerOfAttorney.select(:file_number, :claimant_participant_id).group(:file_number, :claimant_participant_id).having("count(*) > 1").map.count
```

**Comment by pkarman** [comment]
There are 33 duplicate PIDs:

```ruby
BgsPowerOfAttorney.select(:claimant_participant_id)
.group(:claimant_participant_id)
.having("count(*) > 1")
.map { |m| m.claimant_participant_id }
```

**Comment by pkarman** [comment]
So maybe we give up in the face of messy data and drop the unique constraint and just rely on the non-unique indices on PID and fn.

**Comment by pkarman** [comment]
Fix suggested in #14517 that would change the BGS lookup logic based on existing vs new records.
